website/public/js/controllers/groupsCtrl.js

```diff
@@ -224,6 +224,9 @@ habitrpg.controller("GroupsCtrl", ['$scope', '$rootScope', 'Shared', 'Groups', '
       if ($scope.query === undefined || $scope.query === null) {
         return false;
       }
+      if (msg.username == undefined) {
+        return false;
```

Comment:
I also have mixed feelings about `typeof msg.username === 'string'`. Especially because:
`if (typeof msg.username == 'string' || myVar instanceof String)` is apparently a better test (reference [this](http://stackoverflow.com/questions/4059147/check-if-a-variable-is-a-string) post) because JavaScript language semantics are so wonderful!
I was thinking that maybe we could use a function from an external helper library: like [this](https://docs.angularjs.org/api/ng/function/angular.isString) one.

I don't think we can trust that $scope.query.text or msg.username is a string even if they exist. But the ONLY time we want the filter function to return true is when $scope.query is defined and when both $scope.query.text and msg.username are strings. So maybe the function should be:

if ($scope.query && angular.isString($scope.query.text) && msg && angular.isString(msg.username)) {
     return msg.user.indexOf($scope.query.text) == 0;
}
return false;

It is a lot of conditionals, but it is exactly what we want (I think).